### PR TITLE
Pawara

### PR DESCRIPTION
**Pull request by PawaraLiyanage**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -54,7 +54,12 @@
 			<artifactId>spring-boot-starter-test</artifactId>
 			<scope>test</scope>
 		</dependency>
-	</dependencies>
+        <dependency>
+            <groupId>org.apache.xbean</groupId>
```
Comment (kavicastelo) [inline pom.xml:58]: Seems this is unnecessary for this PR. If you need to add this dependency please explain why.

File: src/main/java/com/hris/HRIS/controller/AttendanceController.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.hris.HRIS.controller;
+
+import java.util.ArrayList;
+
+public class AttendanceController {
+
+
+    public void getId(){
```
Comment (kavicastelo) [inline src/main/java/com/hris/HRIS/controller/AttendanceController.java:8]: don't need to create a method to get ID. Mongo repository already supplies you with pre-defined methods to handle collection data with ID.

**Review by kavicastelo — CHANGES_REQUESTED** [review]
Complete the control methods and check out the request changes. Recommended to add CRUD operations minimally.

File: src/main/java/com/hris/HRIS/repository/AttendanceRepository.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.hris.HRIS.repository;
+
+import com.hris.HRIS.model.AttendanceModel;
+import org.springframework.data.mongodb.repository.MongoRepository;
+import java.util.Optional;
+
+public interface AttendanceRepository extends MongoRepository<AttendanceModel, String> {
+    Optional<AttendanceModel> findByEmail(String email);
```
Comment (kavicastelo) [inline src/main/java/com/hris/HRIS/repository/AttendanceRepository.java:8]: you cannot use this custom method here. you can only handle custom methods according to your collection model

File: src/main/java/com/hris/HRIS/repository/AttendanceRepository.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+    Optional<AttendanceModel> deleteByEmail(String email);
```
Comment (kavicastelo) [inline src/main/java/com/hris/HRIS/repository/AttendanceRepository.java:10]: same as the `findByEmail` method. if you need to use a method like this, you need to have an email field in your model. In your case, you can use,
- findByInTime
- FindAllByInTime
- deleteByOutTime
- and blah blah...

**Review by kavicastelo — CHANGES_REQUESTED** [review]
need to change repository methods

**Review by PawaraLiyanage — COMMENTED** [review]
Updated

File: src/main/java/com/hris/HRIS/controller/AttendanceController.java (hunk 2 above)
Comment (PawaraLiyanage) [reply src/main/java/com/hris/HRIS/controller/AttendanceController.java]: done

**Review by kavicastelo — APPROVED** [review]
reviewed and approved
`changed the pom.xml` typo conflict.

Make sure to update your local repository before contributing next time
*remove unwanted space and add tab line 57 in pom.xml*

**Pull request merged**